src/utils/getBVAData.js

```diff
@@ -0,0 +1,155 @@
+            {
+                resultData = { ...aliasDataRaw }; // copy over the first alias data (account, timestamp ect)
+                resultData["aliasURI"] = undefined; // is not needed
+            }
+
+            try { // should give a json with data                
+                let aliasData = JSON.parse(aliasDataRaw['aliasURI']);
+                let sources = resultData.sources == undefined ? aliasData.sources : 
+                                                                aliasData.sources.concat(resultData.sources); // collect all sources
+                resultData = { ...resultData, ...aliasData }; // copy over properties (replaces existing props)
+                resultData.sources = sources; // copy sources array back
+                
+                currentAlias = aliasData.extended_data;
+            }
+            catch (e) {
+                if (e instanceof SyntaxError) {
+                    console.log(e);
+                } else {
+                    console.log(e);				
+                }			
+                currentAlias = undefined;
+            }
+        }
+        else currentAlias = undefined;
+
+        maxExtendedData -= 1;
+	}
+
+    // if it contains an asset ID. load the asset data into the object
+    if (resultData && resultData["asset_id"] > 0)
+    {
+        let assetDataRaw;        
+        try { // get the alias from the blockchain
+            assetDataRaw = await api.asset.getAsset(resultData["asset_id"])
+            resultData["asset_data"] = assetDataRaw; // copy over asset properties (replaces existing props)
+        }
+        catch (error) {
+            console.log(error);
+        }
+        
+    /*  work in progress - collect price data
+        let askUrl = Vars.PeerUrl + "burst?requestType=getAskOrders&asset=" + resultData["asset_id"] + "&firstIndex=1&lastIndex=1"
+        let bidUrl = Vars.PeerUrl + "burst?requestType=getBidOrders&asset=" + resultData["asset_id"] + "&firstIndex=1&lastIndex=1"
+        
+        let assetBidDataRaw;
+        try {
+            assetBidDataRaw = await api.asset.getBidOrders(resultData["asset_id"], 1, 1)
+            resultData["bidOrder"] = assetBidDataRaw;
+        }
+        catch (error) {
+            console.log(error);
+        }
+        let assetAskDataRaw;
+        try {
+            assetAskDataRaw = await api.asset.getAskOrders(resultData["asset_id"], 1, 1)
+            resultData["askOrder"] = assetAskDataRaw;
+        }
+        catch (error) {
+            console.log(error);
+        }*/
+    }
+    // itemize the tags into an array
+    if (resultData && resultData["tags"])
+    {
+        var tagArr = resultData["tags"].split(',');
+        for (let i = 0; i < tagArr.length; i++)
+            tagArr[i] = tagArr[i].trim(); // remove whitespace
+        resultData["tags"] = tagArr;
+    }
+    return resultData;
+}
+
+export async function checkHash(fileList, sourcesHashes)
+{
+    var isValidIdx = 0;

```

Comment:
`var` is not used anymore - use `const` or `let`. There are subtle but important differences, especially regarding hoisting and closure